Only use ROOT in celer-g4 when Celeritas uses Geant4

drbenmorgan opened the pull request:
A really minor fix for an configuration corner case - In the situation that Celeritas uses ROOT but not Geant4, dictionary generation for `celer-g4` will fail because it needs the Geant4 headers.

The changes here just move the dictionary generation inside the `CELERITAS_USE_Geant4` conditional so it only happens when both packages are used. The "no Geant4" case is trivial and doesn't need either package, so this should be safe under all combinations.

drbenmorgan committed:
Only use ROOT in celer-g4 when Celeritas uses Geant4

In the situation CELERITAS_USE_Geant4 is false and CELERITAS_USE_ROOT
is true, dictionary generation will be attempted. This relies on
use of Geant4 includes and thus compilation will fail.

Only generate a dictionary/use ROOT in celer-g4 when both ROOT and
Geant4 are used.

sethrj commented on app/celer-g4/CMakeLists.txt:
```diff
@@ -18,48 +18,48 @@ if(CELERITAS_USE_Geant4)
   set(LIBRARIES)
 endif()
 
-if(CELERITAS_USE_ROOT)
```
The change set might be smaller if you just change this line to `CELERITAS_USE_ROOT AND CELERITAS_USE_Geant4`?

drbenmorgan replied:
My thinking was that the `CELERITAS_USE_Geant4` is the fundamental conditional, so seemed logical to put the "use ROOT" inside of this rather than a second dual conditional later. However, if you prefer to keep changesets small, I'll update!

sethrj replied:
True. I'm also partial to reducing the level of indentation 😅

drbenmorgan replied:
That's also a good reason - o.k., will change to the `CELERITAS_USE_ROOT AND CELERITAS_USE_Geant4` form in a fresh commit, which can then be squashed on merge.

drbenmorgan committed:
Keep ROOT conditional separate from Geant4 to reduce indent

Retain behaviour of only generating dictionaries when both ROOT
and Geant4 are in use.

sethrj reviewed (approved):
Thanks!
